pysb/bng.py

```diff
@@ -137,8 +137,16 @@ def _format_action_args(cls, **kwargs):
             Arguments and values to supply to BNG
         """
         if kwargs:
-            action_args = ','.join('%s=>%s' % (k, BngConsole._bng_param(v))
-                                   for k, v in kwargs.items())
+            processed_pars = []
+            for k, v in kwargs.items():
+                bng_param = BngConsole._bng_param(v)
+                if isinstance(bng_param, collections.Mapping):
+                    par_string = f'{k}' + '=>{' + ','.join(f"{k}=>{v}" for k, v in bng_param.items()) + '}'
```

Comment:
Should we perform some validation on the `kwargs` keys and values here to make sure we don't emit invalid BNGL? If the BNG parser fails due to invalid BNGL here the error will probably be hard for the user to make sense of. The keys should be validated against the regex that matches whatever BNG accepts there, but I'm not sure what the value syntax is.